jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -2497,124 +2059,277 @@ ecma_builtin_array_prototype_object_find (ecma_value_t this_arg, /**< this argum
+    {
+      ret_value = ecma_builtin_array_prototype_object_index_of (routine_arg_1,
+                                                                routine_arg_2,
+                                                                obj_p,
+                                                                length);
+      break;
+    }
+    case ECMA_ARRAY_PROTOTYPE_LAST_INDEX_OF:
+    {
+      ret_value = ecma_builtin_array_prototype_object_last_index_of (arguments_list_p,
+                                                                     arguments_number,
+                                                                     obj_p,
+                                                                     length);
+      break;
+    }
+    case ECMA_ARRAY_PROTOTYPE_EVERY:
+    case ECMA_ARRAY_PROTOTYPE_SOME:
+    case ECMA_ARRAY_PROTOTYPE_FOR_EACH:
+    {
+      ret_value = ecma_builtin_array_apply (routine_arg_1,
+                                            routine_arg_2,
+                                            (array_routine_mode) builtin_routine_id - ECMA_ARRAY_PROTOTYPE_EVERY,
+                                            obj_p,
+                                            length);
+      break;
+    }
+    case ECMA_ARRAY_PROTOTYPE_MAP:
+    {
+      ret_value = ecma_builtin_array_prototype_object_map (routine_arg_1,
+                                                           routine_arg_2,
+                                                           obj_p,
+                                                           length);
+      break;
+    }
+    case ECMA_ARRAY_PROTOTYPE_FILTER:
+    {
+      ret_value = ecma_builtin_array_prototype_object_filter (routine_arg_1,
+                                                              routine_arg_2,
+                                                              obj_p,
+                                                              length);
+      break;
+    }
+    case ECMA_ARRAY_PROTOTYPE_REDUCE:
+    case ECMA_ARRAY_PROTOTYPE_REDUCE_RIGHT:
+    {
+      ret_value = ecma_builtin_array_reduce_from (routine_arg_1,
+                                                  routine_arg_2,
+                                                  arguments_number,
+                                                  builtin_routine_id == ECMA_ARRAY_PROTOTYPE_REDUCE,
+                                                  obj_p,
+                                                  length);
+      break;
+    }
+#ifndef CONFIG_DISABLE_ES2015_BUILTIN
+    case ECMA_ARRAY_PROTOTYPE_FIND:
+    {
+      ret_value = ecma_builtin_array_prototype_object_find (routine_arg_1,
+                                                            routine_arg_2,
+                                                            obj_p,
+                                                            length);
+      break;
+    }
+#endif /* !CONFIG_DISABLE_ES2015_BUILTIN */
+    default:
+    {
+      JERRY_UNREACHABLE ();
```

Comment:
AFAIK this is a bad practice and we started to refactor these unnecessary `JERRY_UNREACHABLE` calls. Please check #2342 for details. CC: @akosthekiss

Reply:
Sure, it can be simply eliminated.